services/api-server/src/simcore_service_api_server/_service_solvers.py

```diff
@@ -0,0 +1,74 @@
+from typing import Annotated
+
+from common_library.pagination_tools import iter_pagination_params
+from fastapi import Depends
+from models_library.basic_types import VersionStr
+from models_library.products import ProductName
+from models_library.rest_pagination import MAXIMUM_NUMBER_OF_ITEMS_PER_PAGE
+from models_library.services_enums import ServiceType
+from models_library.services_history import ServiceRelease
+from models_library.users import UserID
+from packaging.version import Version
+
+from .models.schemas.solvers import Solver, SolverKeyId
+from .services_rpc.catalog import CatalogService
+
+DEFAULT_PAGINATION_LIMIT = MAXIMUM_NUMBER_OF_ITEMS_PER_PAGE - 1
+
+
+class SolverService:
+    _catalog_service: CatalogService
+
+    def __init__(
+        self, catalog_service: Annotated[CatalogService, Depends(CatalogService)]
+    ):
+        self._catalog_service = catalog_service
+
+    async def get_solver(
+        self,
+        *,
+        user_id: UserID,
+        name: SolverKeyId,
+        version: VersionStr,
+        product_name: ProductName,
+    ) -> Solver:
```

Comment:
NOTE: take into account that `Solver` is a schema and not a domain model. For now it is enough refactoring but at some point we should also address this issue.